### PR TITLE
Tp/propensity

### PR DESCRIPTION
**Pull request by tom-parker**
cc @commuterjoy your flag will need to look something like this -

```
    {
        "name": "hasPropensity Test",
        "description": "Test.",
        "owner": "matt.chadburn@ft.com",
        "safeState": false,
        "state": false,
        "tags": [
        ],
        "cohorts": [
            "anonymous"
        ],
        "expiry": "2030-01-01T10:00:00.000Z",
        "abTestState": true,
        "abTestSetup": {
            "offset": 0,
            "range": 100,
            "filters": [
            	{
            		"type": "hasPropensity",
            		"value": true
            	}
            ]
        }
    }
```

### CODE REVIEW

**Comment by commuterjoy** [comment]
is this ready for review? or are you still fixing tests?

**Comment by tom-parker** [comment]
It's ready for review.

I think CI is blowing up because there is no HUI API key and the app will throw if there isn't one. The wheels are in motion to get one, but still waiting.

File: server/filters/haspropensity.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	throw new Error('You need to set PROPENSITY_API_KEY in your environment');
+}
+
+const metrics = require('ft-next-express').metrics;
+
+const userIdPattern = /SIVISITOR=([^;]*)/;
+
+const getAnonymousSessionId = (cookie) => {
+	var matches = userIdPattern.exec(cookie) || [];
+	if (matches) {
```
Comment (commuterjoy) [inline server/filters/haspropensity.js]: matches will never return false/undefined as you've got a `|| []` on the preceding line

**Comment by commuterjoy** [comment]
+1

**Pull request merged**